os/kernel/log_dump/log_dump.c

```diff
@@ -109,10 +140,61 @@ int log_dump_init(void)
 	return LOG_DUMP_OK;
 }
 
-void log_dump_read_init(void)
+unsigned char *last_comp_block;
+size_t last_comp_block_size;
+int last_comp_block_ptr;
+bool last_block_compression;
+
+int log_dump_read_init (void)
+{
+	last_block_compression = true;
+	struct log_dump_msg_s msg;
+
+	msg.comp = true;
+	msg.ch = 'c';
+	mq_send(log_dump_get_mqfd(), (const char *)&msg, sizeof(struct log_dump_msg_s), 100);
+
+	while (last_block_compression);
+	return 0;
+}
+
+int log_dump_read_init_helper(void)
 {
 	read_node = (struct log_dump_chunk_s *)sq_peek(&log_dump_chunks);	/* reset the read node to head */
-	current_ptr = 0;							/* reset the read pointer */
+	if (read_node->comp_head < 0)
+		current_ptr = curbytes;
+	else
+		current_ptr = read_node->comp_head;
```

Comment:
Is it guaranteed that `read_node` cannot be NULL?
If not, it can be dereferencing.